include/bluetooth/gatt.h

```diff
@@ -181,6 +181,22 @@ struct bt_gatt_include {
 	uint16_t end_handle;
 };
 
+/** @brief GATT callback structure. */
+struct bt_gatt_cb {
+	/** @brief The maximum ATT MTU on a connection has changed.
+	 *
+	 *  This callback notifies the application that the maximum TX or RX
+	 *  ATT MTU has increased.
+	 *
+	 *  @param conn Connection object.
+	 *  @param tx Updated TX ATT MTU.
+	 *  @param rx Updated RX ATT MTU.
+	 */
+	void (*att_mtu_updated)(struct bt_conn *conn, uint16_t tx, uint16_t rx);
```

Comment:
How does this fixes any problem that the likes of bt_gatt_get_mtu wouldn't? If there is a race with the MTU change you would likely have this values outdated on the application side.

Reply:
When do you call `bt_gatt_get_mtu` then? You have no way of knowing when the value is likely to change.

Can you expand on the MTU race condition? In my mind this is inherently much safer because instead of polling `bt_gatt_get_mtu`, where the value could change 1 cycle after the call, you are only notified after the change. If the MTU only changes once on a connection like it should, I'm not sure how thats racey.